### PR TITLE
fix some bugs

### PR DESCRIPTION
**Pull request by AlexLiu-Chata**

### CODE REVIEW

File: src/components/Charts/ChataChart/ChataChart.js [hunk 1]
```diff
@@ -541,7 +541,13 @@ export default class ChataChart extends Component {
...
           {!!currencySelectorState.length && (
             <Fragment>
-              <div className="number-selector-header">Currency</div>
+              <div className="number-selector-header">
+                {this.props.dataConfig && this.props.tableColumns
+                  ? this.props.tableColumns[
+                      this.props.dataConfig.legendColumnIndex
+                    ].display_name
+                  : 'Currency'}
+              </div>
```
Comment (AlexLiu-Chata) [inline src/components/Charts/ChataChart/ChataChart.js:550]: AC-992

File: src/components/Charts/ChataChart/ChataChart.js [hunk 2]
```diff
@@ -575,7 +581,14 @@ export default class ChataChart extends Component {
...
             <Fragment>
-              <div className="number-selector-header">Quantity</div>
+              <div className="number-selector-header">
+                {' '}
+                {this.props.dataConfig && this.props.tableColumns
+                  ? this.props.tableColumns[
+                      this.props.dataConfig.legendColumnIndex
+                    ].display_name
+                  : 'Quantity'}
+              </div>
```
Comment (AlexLiu-Chata) [inline src/components/Charts/ChataChart/ChataChart.js:591]: AC-992

File: src/components/Charts/ChataChart/ChataChart.js [hunk 3]
```diff
@@ -608,7 +621,14 @@ export default class ChataChart extends Component {
...
             <Fragment>
-              <div className="number-selector-header">Ratio</div>
+              <div className="number-selector-header">
+                {' '}
+                {this.props.dataConfig && this.props.tableColumns
+                  ? this.props.tableColumns[
+                      this.props.dataConfig.legendColumnIndex
+                    ].display_name
+                  : 'Ratio'}
+              </div>
```
Comment (AlexLiu-Chata) [inline src/components/Charts/ChataChart/ChataChart.js:631]: AC-992

File: src/components/Charts/ChataLineChart/ChataLineChart.js [hunk 4]
```diff
@@ -133,7 +133,7 @@ export default class ChataLineChart extends Component {
         <Axes
           themeConfig={themeConfig}
           scales={{ xScale, yScale }}
-          xCol={columns[stringColumnIndex]}
+          xCol={columns[0]}
```
Comment (AlexLiu-Chata) [inline src/components/Charts/ChataLineChart/ChataLineChart.js:136]: AC-979

File: src/components/Dashboard/DashboardTile/DashboardTile.js [hunk 5]
```diff
@@ -1167,6 +1167,18 @@ export default class DashboardTile extends React.Component {
...
+    )
+    if (dashboardTileToolTipElement) {
+      if (_get(this.props, 'style.width')) {
+        dashboardTileToolTipElement.style.maxWidth = `${_get(
+          this.props,
+          'style.width'
+        ).match(/\d+/g)[0] - 75}px`
+      }
+    }
+
```
Comment (AlexLiu-Chata) [inline src/components/Dashboard/DashboardTile/DashboardTile.js:1181]: AC-981

File: src/components/Dashboard/DashboardTile/DashboardTile.js [hunk 6]
```diff
@@ -1152,7 +1152,7 @@ export default class DashboardTile extends React.Component {
           type="warning"
           className="dashboard-data-limit-warning-icon"
           data-tip={`The display limit of ${numRows} rows has been reached. Try querying a smaller time-frame to ensure all your data is displayed.`}
-          data-for="chart-element-tooltip"
+          data-for="dashboard-data-limit-warning-tooltip"
```
Comment (AlexLiu-Chata) [inline src/components/Dashboard/DashboardTile/DashboardTile.js:1155]: AC-981

File: src/components/Dashboard/Dashboard.js [hunk 7]
```diff
@@ -757,6 +757,13 @@ class Dashboard extends React.Component {
             delayShow={500}
             html
           />
+          <ReactTooltip
+            place="left"
+            className="react-autoql-chart-tooltip"
+            id="dashboard-data-limit-warning-tooltip"
+            effect="solid"
+            html
+          />
```
Comment (AlexLiu-Chata) [inline src/components/Dashboard/Dashboard.js:766]: AC-981

**Review by Ben-CHATA — APPROVED** [review]

**Pull request merged**